Streamline install in `.travis.yml`

hdoupe opened the pull request:
This PR streamlines the CCC installation process. This PR uses the `environment.yml` file to build CCC's requirements. I removed `requirements.txt` because it seemed like it wasn't used anymore. I also removed the `conda-requirements.txt` file because it is redundant to the `environment.yml` file. 

I also dropped a couple environment variables from `.travis.yml` and removed the commented out lines of code that referenced them. I'm happy to add them back if I've missed something.

I ran into some errors when running the tests locally from the `test_4package.py` script. I'm not really sure what that does. I can remove that in this PR if that's helpful.

jdebacker commented:
@hdoupe I appreciate these changes.  They all look good. 

Should the `compdevkit` be a dependency given the tests in `.comp/tests/test_functions.py`?

It seems that `test_4package.py` is passing on Travis CI.  I don't know how useful that test is, so feel free to delete if you don't see value in it.

hdoupe commented:
I added `compdevkit`. I'll have a proper pypi package up this week, and we can drop the git install at that point. 

I dropped the `test_4package.py` script.
